packages/babel-plugin-transform-async-to-generator/src/index.js

```diff
@@ -9,6 +9,17 @@ export default function () {
       Function(path, state) {
         if (!path.node.async || path.node.generator) return;
 
+        // Ensure any bindings at the Program level are renamed first
+        // so any further renames cannot accidentally rename the Promise
+        // binding used by the inline helper
+        const programScope = path.scope.getProgramParent();
+        if (programScope.hasBinding("Promise", true)) {
+          programScope.rename("Promise");
```

Comment:
This was a bit of a brute force approach, but was required to make the `shadowed-promise-nested` test case pass.